src/scipp/format/formatter.py

```diff
@@ -27,53 +24,65 @@ def format_variable(data, spec):
     var = data.variances if data.shape else array((data.variance, ))
     unt = "" if data.unit == Unit('dimensionless') else f" {data.unit}"
 
-    if compact and not (var is None or any([x is None for x in var])):
-        from numpy import sqrt
-        return f"{_format_to(*_round_to(val, sqrt(var)))}{unt}"
-    elif compact:
-        return f"{', '.join([f'{v}' for v in val])}{unt}"
+    if compact:
+        # Iterate over array values to handle no- and infinite-precision cases
+        if var is None:
+            formatted = [_format(v) for v in val]
+        else:
+            formatted = [_format(*_round(v, e)) for v, e in zip(val, var)]
+        return f"{', '.join(formatted)}{unt}"
+        # return f"{_format(val, var)}{unt}"
 
     # punt (for now)
     return data.__repr__()
 
 
-def _round_to(values: ndarray, errors: ndarray):
-    from numpy import floor, log10, round, power, any
+def _round(value, variance):
+    from numpy import floor, log10, round, power, sqrt
+    # Treat 'infinite' precision the same as no variance
+    if variance is None or variance == 0:
```

Comment:
How can `variance` be `None` here? Did you intend to call thins function also when there is no variance?

Reply:
It's in the scalar case, where
```
    var = data.variances if data.shape else array((data.variance, ))
```
gives an `array([None], dtype=object)`.

I can remove the `None` check in `_round` if I modify the above line to
```
    var = data.variances if data.variances is None or data.shape else array((data.variance,))
```
but maybe that is too convoluted to actually use.

Reply:
Leave it